src/components/InngestCommHandler.ts

```diff
@@ -748,8 +753,63 @@ export class InngestCommHandler<H extends Handler, TransformedRes> {
     return this.showLandingPage ?? strBoolean(strEnvVar) ?? true;
   }
 
-  protected validateSignature(): boolean {
-    return true;
+  protected validateSignature(sig: string | undefined, body: Record<string, any>) {
+    if (!this.signingKey) {
+      console.warn(
+        "No signing key provided to validate signature.  Find your dev keys at https://app.inngest.com/test/secrets"
+      );
+      return;
+    }
+
+    if (this.isProd && !sig) {
+      throw new Error("No x-inngest-signature provided");
+    }
+
+    if (!sig) {
+      console.warn("No x-inngest-signature provided");
+      return;
+    }
+
+    // sig should have a format of `t=${unixTS}&s=${signature}` to parse.
+    //
+    // We cant use URLSearchParams here because this needs to run directly in V8
```

Comment:
`URLSearchParams` should be in all the major envs including v8-like envs like Deno + Cloudflare Workers (https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#browser_compatibility) and I believe it's a global in Node since v10. This solution works, but if we want to use a built-in parser we should be able to.

Reply:
IIRC it had issues with cloudflare workers, so people had workarounds: https://community.cloudflare.com/t/parse-url-query-strings-with-cloudflare-workers/90286.

Reply:
Odd, looked like from 2020, I would have hoped they fixed that by now! 😬